### PR TITLE
Add YouTube OAuth2 flow

### PR DESCRIPTION
**Pull request by evan10s**
- Backend and frontend changes to allow connecting a YouTube channel using OAuth2
  - Added concept of secrets to the backend. These behave similar to settings, except that secrets aren't included in the settings API endpoint
  - Once the initial YouTube connection is complete, the client ID and secret can't be changed (from the frontend) since this will break YouTube authentication
  - Server calculates redirect URI from request metadata and environment variables (`HOST`) and the frontend displays the dynamic value
- Since I needed to add a concept of non-environment variable secrets, I generalized `SecretsRepo` to `JsonStorageRepo`.
  - The only difference between a setting and a secret is that secret values cannot be access via any HTTP routes
  - To ease getting started, the server will silently copy the template (`.example.json`) settings/secrets files to their permanent versions (`settings.json`/`secrets.json`) so users don't have to worry about creating these files.
- Added linting and running server unit tests to continuous integration with Github Actions

TODO:
- [x] Finish cleaning up CI
- [x] Confirm everything still works as expected after all the style fixes

Screenshots
- Overall flow GIF (note that this looks slightly different, see the screenshot of the redirect URI field below)
full-youtube-flow

- Providing the redirect URI value:
image

- Error handling
image

### CODE REVIEW

File: client/src/views/settings/Settings.vue [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        </VForm>
+
+        <h2>YouTube</h2>
+
+        <h3 class="mb-2">OAuth2 client details</h3>
+        <VAlert class="mb-3"
+                v-if="!youTubeAuthState?.accessTokenStored"
+        >In your Google Cloud project, create an OAuth2 web client.<br />
+          <br />
+          Be sure to add <code>http://localhost:3000/api/v1/youtube/auth/callback</code> as an authorized redirect.
```
Comment (RossIV) [inline client/src/views/settings/Settings.vue:47]: Referencing `localhost` is fine here for now, but I could see a future where we'd want to populate that with whatever hostname was used in the browser to load the page. Or failing that, explicitly decide that the YouTube pairing process can only happen from the machine running the uploader but I'm not as much of a fan of that.

File: server/src/routes/settings.ts
Comment (RossIV) [inline server/src/routes/settings.ts:1]: Lots of warnings and some errors here - are these of concern?

File: server/src/routes/youtube.ts
Comment (RossIV) [inline server/src/routes/youtube.ts:1]: Lots of warnings and some errors here - are these of concern?

File: server/src/services/SettingsService.ts
Comment (RossIV) [inline server/src/services/SettingsService.ts:1]: Lots of warnings and some errors here - are these of concern?

File: server/src/services/YouTubeService.ts
Comment (RossIV) [inline server/src/services/YouTubeService.ts:1]: Lots of warnings and some errors here - are these of concern?

File: client/src/views/settings/Settings.vue (hunk 1 above)
Comment (evan10s) [reply client/src/views/settings/Settings.vue:47]: It should probably pull the hostname/domain of the current page and use that as the suggestion. But technically you can already set it to whatever you please, although it requires manually adjusting the value in settings.json

File: server/src/routes/settings.ts
Comment (khari05) [reply server/src/routes/settings.ts:1]: We could just run `eslint --fix` and it should get rid of most of them

Comment (evan10s) [reply server/src/routes/settings.ts:1]: Yes, I'm intending to make the rules more agreeable (probably 120 for max line length).

I also have a perhaps controversial opinion that warnings are useless in this context - it's either a problem or it isn't. getting the CI working in itself was enough of a challenge (so I took a break afterwards), and I could `--fix` everything but there's a balance of not creating a lot of whitespace changes in this PR. but basically, i am intending to reduce the warnings to 0 either by adjusting rules or making them errors

Comment (evan10s) [reply server/src/routes/settings.ts:1]: Fixed up all the style stuff

File: client/src/views/settings/Settings.vue (hunk 1 above)
Comment (evan10s) [reply client/src/views/settings/Settings.vue:48]: I fixed this up a little bit so it now uses `window.location.origin` which should give us everything before the path on the domain (e.g., "http://localhost:3000")

I also added a fancy copy button for the URL and the text input will select all its text when you focus it:
image

**Pull request merged**